Automattic/jetpack
# Performance: VideoPress: avoid useEffect call for every block on typing

Fixes #

## Proposed changes:

* We should return early before `useEffect` is called. Currently this effect is running for every single block on the page on every single keystroke.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






### Test `<JetpackCoreVideoDeprecation />` block

The app filters all blocks and will extend only the `core/video` videopress  instances.

* Go to the block editor
* Create a few general block instances, like paragraphs
* Ensure the VideoPress video block is not supported for your site
  * Deactive the Jetpack module: `https://your-testing-site/wp-admin/admin.php?page=jetpack#/settings?term=video`
<img width="576" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/32444eb0-bf1b-4f0a-b859-bba20ee3fb1c">
  * Deactivate the VideoPress standalone plugin 
 
<img width="593" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/05d06430-bc21-4581-9674-76cc187f5af8">

* Open the React Dev tool. Search by `ConvertV6toV5Effect` string
* Confirm there isn't any `JetpackCoreVideoDeprecation` instance in the components three

* Create a `videopress/video` block instance
  * Switch the editor mode
  * Paste the following chunk of code:
 
```

```

  * Return to visual mode
  * Confirm the editor identifies the new block like a `core/video` block instance
  * Confirm the block sidebar shows the transform info section

<img width="945" alt="Screenshot 2023-10-21 at 12 39 04" src="https://github.com/Automattic/jetpack/assets/77539/4495e601-27f2-49a5-b402-29f3bf414635">
